media/js/firefox/australis/australis-uitour.js

```diff
@@ -56,14 +56,71 @@ if (typeof Mozilla == 'undefined') {
         return id;
     }
 
+    var notificationListener = null;
+    function _notificationListener(event) {
+        if (typeof event.detail != 'object')
+            return;
+        if (typeof notificationListener != 'function')
+            return;
+
+        notificationListener(event.detail.event, event.detail.params);
+    }
+
     Mozilla.UITour.DEFAULT_THEME_CYCLE_DELAY = 10 * 1000;
 
+    Mozilla.UITour.CONFIGNAME_SYNC = 'sync';
+    Mozilla.UITour.CONFIGNAME_AVAILABLETARGETS = 'availableTargets';
+
+    Mozilla.UITour.ping = function(callback) {
+        var data = {};
+        if (callback) {
+            data.callbackID = _waitForCallback(callback);
+        }
+        _sendEvent('ping', data);
+    };
+
+    Mozilla.UITour.observe = function(listener, callback) {
+        notificationListener = listener;
+
+        if (listener) {
+            document.addEventListener('mozUITourNotification',
+                                                                _notificationListener);
+            Mozilla.UITour.ping(callback);
+        } else {
+            document.removeEventListener('mozUITourNotification',
+                                                                     _notificationListener);
+        }
+    };
+
     Mozilla.UITour.registerPageID = function(pageID) {
         _sendEvent('registerPageID', {
             pageID: pageID
         });
     };
 
+    Mozilla.UITour.showHeartbeat = function(message, thankyouMessage, flowId, engagementURL,
```

Comment:
Ah, good advice. I was a little concerned at how much was added here. Didn't know we were already picking and choosing. :smiley: